core/src/test/java/org/elasticsearch/threadpool/GenericThreadPoolTests.java

```diff
@@ -0,0 +1,210 @@
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasToString;
+
+public class GenericThreadPoolTests extends ESThreadPoolTestCase {
+
+    public void testGenericThreadPoolConfiguration() throws InterruptedException {
+        final Settings.Builder builder = Settings.builder();
+        final int max;
+        if (randomBoolean()) {
+            final int processors = randomIntBetween(1, 64);
+            builder.put("processors", processors);
+            max = ThreadPool.boundedBy(4 * EsExecutors.boundedNumberOfProcessors(builder.build()), 128, 512);
+        } else {
+            max = ThreadPool.boundedBy(Math.min(32, Runtime.getRuntime().availableProcessors()), 128, 512);
+        }
+
+        final long keepAlive;
+        if (randomBoolean()) {
+            keepAlive = randomIntBetween(1, 300);
+            builder.put("threadpool.generic.keep_alive", keepAlive + "s");
+        } else {
+            keepAlive = 30; // the default;
+        }
+
+        runGenericThreadPoolTest(builder.build(), (clusterSettings, threadPool) -> {
+            final Executor executor = threadPool.generic();
+            assertThat(executor, instanceOf(EsThreadPoolExecutor.class));
+            final EsThreadPoolExecutor generic = (EsThreadPoolExecutor)executor;
+            final ThreadPool.Info info = info(threadPool, "generic");
+
+            assertThat(info.getName(), equalTo("generic"));
+            assertThat(info.getThreadPoolType(), equalTo(ThreadPool.ThreadPoolType.SCALING));
+
+            assertThat(info.getKeepAlive().seconds(), equalTo(keepAlive));
+            assertThat(generic.getKeepAliveTime(TimeUnit.SECONDS), equalTo(keepAlive));
+
+            assertNull(info.getQueueSize());
+            assertThat(generic.getQueue().remainingCapacity(), equalTo(Integer.MAX_VALUE));
+
+            assertThat(info.getMin(), equalTo(4));
+            assertThat(generic.getCorePoolSize(), equalTo(4));
+            assertThat(info.getMax(), equalTo(max));
+            assertThat(generic.getMaximumPoolSize(), equalTo(max));
+        });
+    }
+
+    public void testValidDynamicKeepAlive() throws InterruptedException {
+        runGenericThreadPoolTest(Settings.EMPTY, (clusterSettings, threadPool) -> {
+            final Executor beforeExecutor = threadPool.generic();
+            final long seconds = randomIntBetween(1, 300);
+            clusterSettings.applySettings(settings("threadpool.generic.keep_alive", seconds + "s"));
+            final Executor afterExecutor = threadPool.generic();
+            assertSame(beforeExecutor, afterExecutor);
+            final ThreadPool.Info info = info(threadPool, "generic");
+            assertThat(info.getKeepAlive().seconds(), equalTo(seconds));
+        });
+    }
+
+    public void testGenericThreadPoolIsBounded() throws InterruptedException {
```

Comment:
Sure, that's an artifact how initially the generic thread pool was the only cached thread pool (and thus separate from scaling) but during the evolution of this PR the cached thread pool was removed and the generic thread pool because just another scaling thread pool.